src/ctia/lib/kafka.clj

```diff
@@ -0,0 +1,119 @@
+(ns ctia.lib.kafka
+  (:require
+   [clojure.tools.logging :as log]
+   [ctia.properties :refer [properties]]
+   [onyx.kafka.helpers :as okh]
+   [onyx.plugin.kafka :as opk])
+  (:import kafka.admin.AdminUtils
+           kafka.admin.AdminClient))
+
+(defn decompress
+  [v]
+  (when v (String. v "UTF-8")))
+
+(defn ssl-enabled? []
+  (boolean (get-in @properties
+                   [:ctia :hook :kafka :ssl :enabled])))
+
+(defn make-ssl-opts []
+  (let [ssl-props (get-in @properties [:ctia :hook :kafka :ssl])]
+    {"security.protocol" "ssl"
+     "ssl.truststore.location"
+     (get-in ssl-props [:truststore :location])
+     "ssl.truststore.password"
+     (get-in ssl-props [:truststore :password])
+     "ssl.keystore.location"
+     (get-in ssl-props [:keystore :location])
+     "ssl.keystore.password"
+     (get-in ssl-props [:keystore :password])
+     "ssl.key.password"
+     (get-in ssl-props [:key :password])}))
+
+(defn build-producer []
+  (let [producer-opts {}
+        {:keys [request-size compression]}
+        (get-in @properties [:ctia :hook :kafka])
+        compression-type (:type compression)
+        {:keys [session-timeout
+                connection-timeout
+                operation-retry-timeout
+                address] :as zk-config}
+        (get-in @properties [:ctia :hook :kafka :zk])
+        brokers (opk/find-brokers {:kafka/zookeeper address})
+        kafka-config (cond-> {"bootstrap.servers" brokers
+                              "max.request.size" request-size}
+                       (ssl-enabled?) (merge (make-ssl-opts))
+                       compression-type (assoc "compression.type"
+                                               compression-type))]
+    (okh/build-producer kafka-config
+                        (okh/byte-array-serializer)
+                        (okh/byte-array-serializer))))
+
+(defn build-consumer []
+  (let [{:keys [request-size]}
+        (get-in @properties [:ctia :hook :kafka])
+        {:keys [session-timeout
+                connection-timeout
+                operation-retry-timeout
+                address] :as zk-config}
+        (get-in @properties [:ctia :hook :kafka :zk])
+        brokers (opk/find-brokers {:kafka/zookeeper address})
+        kafka-config (cond-> {"bootstrap.servers" brokers
+                              "group.id" "ctia"}
+                       (ssl-enabled?) (merge (make-ssl-opts)))]
+    (okh/build-consumer kafka-config
+                        (okh/byte-array-deserializer)
+                        (okh/byte-array-deserializer))))
+
+(defn poll [consumer name f timeout]
```

Comment:
It would be nice to have a docstring here. It is not perfectly clear to me what `consumer`, `name`, `f` and `timeout` are here.

In particular it seems that `f` is a handler so it should be preferable to rename it `handler` instead or `handle-event`.